server/account/account.go

```diff
@@ -13,8 +17,65 @@ const (
 
 type AccountID [HashSize]byte
 
-func New(pk [pki.PubKeySize]byte) AccountID {
+// NewID generates a unique account id with the provided public key bytes.
+func NewID(pk [pki.PubKeySize]byte) AccountID {
 	// hash the pubkey hash(hash(pubkey))
 	h := HashFunc(pk[:])
 	return HashFunc(h[:])
 }
+
+// Account represents a dex account.
+type Account struct {
+	ID      AccountID
+	PrivKey *secp256k1.PrivateKey
+}
+
+// NewAccount creates a new dex account. The associated private key of the
+// account is generated randomly.
+func NewAccount() (*Account, error) {
+	privKey, err := secp256k1.GeneratePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey := privKey.PubKey()
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:pki.PubKeySize])
+	return &Account{
+		ID:      NewID(pkb),
+		PrivKey: privKey,
+	}, nil
+}
+
+// LoadFromBytes initializes a dex account using the provided private
+// key bytes.
+func (acc *Account) LoadFromBytes(privK []byte) error {
```

Comment:
Would this be better as an alternative constructor rather than a method of `Account`?

Reply:
Possibly, making modifications and adding tests. Will factor that in.